.github/actions/clp-core-build-containers/action.yaml

```diff
@@ -35,58 +27,46 @@ runs:
         username: "${{github.actor}}"
         password: "${{inputs.token}}"
 
-    - id: "get_deps_image_props"
+    - id: "get_image_props"
       shell: "bash"
       run: |
+        name="clp-core-dependencies-x86-${{inputs.os_name}}"
         if [[ "${{inputs.push_deps_image}}" == "true" ]] ; then
-          echo "REGISTRY=ghcr.io" >> "$GITHUB_OUTPUT"
-          echo "BRANCH=${{github.ref_name}}" >> "$GITHUB_OUTPUT"
+          # Docker doesn't support repository names with uppercase characters, so we convert to
+          # lowercase here.
+          repository=$(echo '${{github.repository}}' | tr '[:upper:]' '[:lower:]')
+
+          echo "full_name=ghcr.io/${repository}/${name}" >> "$GITHUB_OUTPUT"
+          echo "branch=${{github.ref_name}}" >> "$GITHUB_OUTPUT"
+          echo "output=type=registry" >> "$GITHUB_OUTPUT"
         else
-          echo "REGISTRY=localhost:${{inputs.local_registry_port}}" >> "$GITHUB_OUTPUT"
-          echo "BRANCH=latest" >> "$GITHUB_OUTPUT"
+          image_path="/tmp/image.tar"
+          echo "image_path=${image_path}" >> "$GITHUB_OUTPUT"
+
+          echo "full_name=${name}" >> "$GITHUB_OUTPUT"
+          echo "branch=latest" >> "$GITHUB_OUTPUT"
+          echo "output=type=docker,dest=${image_path}" >> "$GITHUB_OUTPUT"
         fi
-        # Docker doesn't support repository names with uppercase characters, so we convert to
-        # lowercase here.
-        echo "REPOSITORY=$(echo '${{github.repository}}' | tr '[:upper:]' '[:lower:]')" \
-          >> "$GITHUB_OUTPUT"
 
     - id: "deps_image_meta"
       uses: "docker/metadata-action@v5"
       with:
-        images: "${{steps.get_deps_image_props.outputs.REGISTRY}}\
-          /${{steps.get_deps_image_props.outputs.REPOSITORY}}\
-          /clp-core-dependencies-x86-${{inputs.os_name}}"
+        images: "${{steps.get_image_props.outputs.full_name}}"
         tags: |
-          type=raw,value=${{steps.get_deps_image_props.outputs.BRANCH}}
+          type=raw,value=${{steps.get_image_props.outputs.branch}}
 
     - uses: "docker/build-push-action@v5"
       with:
         context: "${{inputs.clp_core_dir}}"
         file: "${{inputs.clp_core_dir}}/tools/docker-images/clp-env-base-${{inputs.os_name}}\
           /Dockerfile"
-        push: true
-        tags: "${{steps.deps_image_meta.outputs.tags}}"
         labels: "${{steps.deps_image_meta.outputs.labels}}"
+        tags: "${{steps.deps_image_meta.outputs.tags}}"
+        outputs: "${{steps.get_image_props.outputs.output}}"
 
-    - uses: "./.github/actions/clp-core-build"
```

Comment:
Why is ./.github/actions/clp-core-build no longer needed?

Reply:
Previously we were basically chaining (1) building the container, (2) building clp and running its unit tests, and (3) building and pushing the container image containing the clp binaries using actions.

Now we're chaining them using jobs instead, so (2) and (3) are different jobs in the main workflow.